pkg/ddevapp/ddevapp.go

```diff
@@ -389,22 +389,28 @@ func (app DdevApp) GetDocroot() string {
 	return app.Docroot
 }
 
-// GetComposerRoot will determine the absolute composer root directory where
-// all Composer related commands will be executed.
-// If inContainer set to true, the absolute path in the container will be
-// returned, else the absolute path on the host.
-// If showWarning set to true, a warning containing the composer root will be
-// shown to the user to avoid confusion.
-func (app *DdevApp) GetComposerRoot(inContainer, showWarning bool) string {
-	basePath := ""
+// GetAbsDocroot returns the absolute path to the docroot on the host or if
+// inContainer is set to true in the container.
+func (app DdevApp) GetAbsDocroot(inContainer bool) string {
+	var basePath string
 
 	if inContainer {
 		basePath = "/var/www/html"
 	} else {
 		basePath = app.AppRoot
 	}
 
-	absComposerRoot := path.Join(basePath, app.ComposerRoot)
+	return path.Join(basePath, app.GetDocroot())
+}
+
+// GetComposerRoot will determine the absolute composer root directory where
+// all Composer related commands will be executed.
+// If inContainer set to true, the absolute path in the container will be
+// returned, else the absolute path on the host.
+// If showWarning set to true, a warning containing the composer root will be
+// shown to the user to avoid confusion.
+func (app *DdevApp) GetComposerRoot(inContainer, showWarning bool) string {
+	absComposerRoot := path.Join(app.GetAbsDocroot(inContainer), app.ComposerRoot)
```

Comment:
This logic isn't correct; The composer_root is not relative to the docroot. So this ends up with the wrong things in a basic craft3 download, where the composer.json is in the project root (where it should be)

Reply:
Oh it is correct, it has nothing changed here, but the name of the new function is wrong, should be the absolute project root instead.

Reply:
True, messed up sth with the refactoring.